packages/store/src/utils/store-validators.ts

```diff
@@ -47,4 +48,16 @@ export abstract class StoreValidators {
 
     return meta;
   }
+
+  public static checkExtensibleBeforeDecorate(target: object, name: NGXS_DECORATOR): void {
+    if (!Object.isExtensible(target)) {
```

Comment:
Isn't this check redundant? I mean who does `preventExtensions` on the component? I've never seen this:
```ts
@Component({ ... })
class AppComponent {
  constructor() {
    Object.preventExtensions(this);
  }
}
```

Reply:
@markwhitfeld what do you think?

Reply:
Sorry, I also don't think that these checks add anything useful.

Reply:
done